client/plots/disco_new/mapper/CnvArcsMapper.ts

```diff
@@ -0,0 +1,190 @@
+
+        this.cnvClassMap.set(CnvType.Gain, gain)
+        this.cnvClassMap.set(CnvType.Loss, loss)
+        this.cnvClassMap.set(CnvType.Cap, cap)
+
+        this.maxAbsValue = Math.max(Math.abs(this.capMaxValue(cnvMinValue)), Math.abs(this.capMaxValue(cnvMaxValue)))
+    }
+
+    map(arcData: Array<Data>): Array<CnvArc> {
+        const arcs: Array<CnvArc> = []
+
+        arcData.forEach(data => {
+            let startAngle = this.calculateStartAngle(data)
+            let endAngle = this.calculateEndAngle(data)
+
+            if (endAngle - startAngle < this.onePxArcAngle) {
+                const restAngle = this.onePxArcAngle - (endAngle - startAngle)
+                startAngle = startAngle - restAngle / 2
+                endAngle = startAngle + restAngle / 2
+            }
+
+            const innerRadius = this.calculateInnerRadius(data)
+
+            const outerRadius = this.calculateOuterRadius(data)
+
+            const color = this.getColor(data.value)
+
+            const arc = new CnvArc(startAngle,
+                endAngle,
+                innerRadius,
+                outerRadius,
+                color,
+                data.gene,
+                data.chr,
+                data.start,
+                data.stop,
+                data.value,
+                this.cnvUnit
+            )
+
+            arcs.push(arc)
+        })
+
+        return arcs
+    }
+
+    calculateStartAngle(data: Data) {
+        const index = this.reference.chromosomesOrder.indexOf(data.chr)
+        const chromosome = this.reference.chromosomes[index]
+        return chromosome.startAngle + ((chromosome.endAngle - chromosome.startAngle) * (Number(data.start) / chromosome.size));
+    }
+
+    private calculateEndAngle(data: Data) {
+        const index = this.reference.chromosomesOrder.indexOf(data.chr)
+        const chromosome = this.reference.chromosomes[index]
+        return chromosome.startAngle + ((chromosome.endAngle - chromosome.startAngle) * (Number(data.stop) / chromosome.size));
+    }
+
+    getColor(value) {
+
+        const lossColor = this.settings.cnv.lossColor
+        const ampColor = this.settings.cnv.ampColor
+        const cappedLossColor = this.settings.cnv.cappedLossColor
+        const cappedAmpColor = this.settings.cnv.cappedAmpColor
+
+
+        if (value < this.lossCapped) {
+            return cappedLossColor
+        } else if (value <= 0) {
+            return lossColor
+        } else if (value <= this.gainCapped) {
+            return ampColor
+        } else if (value > this.gainCapped  ) {
+            return cappedAmpColor
+        }
+    }
+
+    private calculateInnerRadius(data: Data) {
+        if (this.gainOnly) {
+            return this.settings.rings.cnvInnerRadius
+        }
+
+        if (this.lossOnly) {
+            let outerRadius = this.settings.rings.cnvInnerRadius + this.settings.rings.cnvWidth
```

Comment:
@aacic since the value didn't change within the variable scope, should have declared as `const` instead of `let`

Reply:
This was fixed by the PR:
https://github.com/stjude/proteinpaint/pull/234
--fix option for eslint converts automatically let to const.